### ISSUE TITLE
DBus signalling

### ISSUE THREAD

**Issue by willprice** [popcornmix/omxplayer#571]
It'd be really nice to have omxplayer emit DBus signals for events such as 

* play/pause
* video starts rendering

This would allow library developers to build more robust wrappers around omxplayer.

I'm willing to do the work, but would need guidance as I'm not much a C++ dev and haven't developed for omxplayer before. I've had quite a good poke around in `OMXControl.cpp` and it would be trivial to add signal emissions here, but I'd like to know whether this is the right place to do it? Perhaps the signals should be emitted from the different units that actually perform the work, e.g. `OMXAudio.cpp` etc.

I'd appreciate thoughts from the developers of the project on the suggestion.

Many thanks for your work

**Comment by kennyyy24** [comment]
I agree that it would be a nice functionality to have in omxplayer. I remember leaving a comment about it in OMXcontrol.cpp when I did update the DBus functionalities.

I think you are on the right track for implementation: the core DBus signalling should be performed in OMXControl. The question is to figure out a way to internally pass a signal to OMXControl class from OMXAudio when an event occurs, or, given all the play/pause actions go through the main loop, it could be triggered from there. I'm more skeptical regarding the 'start rendering' event though.

**Comment by bhong2002** [comment]
hello :)

have done playin' around popcornmix omxplayer together with omxplayer-wrapper(python) using python3.4 and i'm happy.... but with a "debian stretch release(9)" .. i got the headache :)

thanks :)

**Comment by willprice** [comment]
@bhong2002 how'd you mean? I'm the author of omxplayer-wrapper hence why I'd like some of these features to improve my library for controlling omxplayer.

**Comment by bhong2002** [comment]
@willprice.. that's why i am here supporting you.. :)... thank you for omxplayer-wrapper.. it is very usefull...
